### ISSUE TITLE
Generated setup.py doesn't install entry points

### ISSUE THREAD

**Issue by fkrull** [takluyver/flit#336]
Because the generated setup.py imports distutils, the metadata for entry points doesn't get installed when doing `python setup.py install`. Instead of a .dist-info directory with an entry-points.txt file, only a .egg-info file gets installed next to the package. Using pip to install from an sdist works because pip performs some black magic to inject setuptools into setup.py.

**Comment by takluyver** [comment]
The generated `setup.py` is meant as a shim for old versions of pip. Pip is able to avoid the issues that installing with setuptools otherwise causes. So it deliberately doesn't use setuptools.

**Comment by fkrull** [comment]
I mean, I understand that. But it means the sdist is not usable with other tools that use sdists but aren't pip. Or more accurately, it is usable but sometimes incorrect in non-obvious ways.

Case in point, I converted a package of mine to flit and then wanted to update the Debian package. Debian's support tools don't support PEP-517 yet (there is specific support for flit which is literally several days old), but I figured it would still work with the setup.py shim. And it did, until I inspected the deb and noticed that any reference to entry points was missing.

If you're against changing this aspect, I think the limitations of the shim should at least be documented explicitly. They were certainly non-obvious to me.
